crates/valence_new/src/inventory.rs

```diff
@@ -104,58 +104,71 @@ impl Inventory {
         std::mem::replace(&mut self.title, title.into())
     }
 
-    pub(crate) fn slot_slice(&self) -> &[Option<ItemStack>] {
+    fn slot_slice(&self) -> &[Option<ItemStack>] {
         self.slots.as_ref()
     }
 }
 
 /// Send updates for each client's player inventory.
-pub(crate) fn update_player_inventories(mut query: Query<(&mut Inventory, &mut Client)>) {
+pub(crate) fn update_player_inventories(
+    mut query: Query<(&mut Inventory, &mut Client), Without<OpenInventory>>,
+) {
     for (mut inventory, mut client) in query.iter_mut() {
         if inventory.kind != InventoryKind::Player {
             warn!("Inventory on client entity is not a player inventory");
         }
 
         if inventory.modified != 0 {
```

Comment:
I think I might be able to combine `update_player_inventories` and `update_open_inventories` but I'm going to wait until this is merged because I don't want to keep delaying this.

Reply:
That looks like something that should happen. Although it might be difficult with the way the queries overlap.